tests/resources/app/resources/database/seeds/2021_03_06_101000_seed_mock_data.cfc

```diff
@@ -0,0 +1,17 @@
+component {
+
+	function up( schema, query ) {
+		var mockData = new testbox.system.modules.mockdatacfc.models.MockData();
```

Comment:
With the understanding that Seeder files do not need to behave like migrations, let's explore some options here:

1. I like the idea of `mockData` being injected by default.  (I would stick to injection if at all possible since `arguments` to `run` will be determined by the Manager.
2. I think the method name could just be `run` since there is no concept of `up` or `down` for seeders.

Reply:
I agree.  The `runMigration` code only knows `up` or `down`, currently so I would have to add some additional switches/conditionals for that.

Reply:
Seeders just follow a command patter, where they just need to `run()`.

Reply:
I also suggest that `mockdatacfc` become a dependency of migrations. I find myself using them not only on seeders but on schema migrations as well.  So I would suggest passing in the mockdata as an argument to the `up` and `down` methods.

Also, for the Seeders, you should pass in the mock data either via a `run( mockdata )` method, or do DI on it and have a `run()` and the `mockdatacfc` injected already.

Reply:
Why would you use Mock data on an actual migration - seems to me that you're using them as seeders then, anyway.

Reply:
I'm curious as well @lmajano.  I can't think of a case for mock data inside a migration.  Can you share an example?

Reply:
I'm going to push back on the MockData injection idea.  It means adding a development tool as a dependency of the module.  While that may not be a big deal, It feels "smelly" to me.  

Two options:

- Add it as a dev dependency to the module, which means that `install --production` wouldn't include it ( or testbox, most likely )
- Keep it as is, and let them inject it in to their seeds, if they want it.  
-

Reply:
Adding it as a dev dependency would do nothing in most cases since you only install your app's dev dependencies, not your dependency's dev dependencies. (whew)  As long as WireBox is creating the seeders it is easy to inject ourselves.  I am fine leaving it out.

Reply:
My preference is to do that and make the seed responsible for its own data.  I can see use cases where we wouldn't be seeding from MockData but might be loading from static JSON files or even pulling data from another database.

Reply:
> I'm curious as well @lmajano. I can't think of a case for mock data inside a migration. Can you share an example?

I have used it to generate strings or random selection of items where I am preparing a db with data.  I used it in ContentBox to add site data when installing for the first time.  The mockdatacfc can generate schemas according to what you want. So not only does it have to be mock data, but it can follow a real data path with structure.

Reply:
I see it as an issue that we don't provide a standard way to mock data in the seeders.  I get that you can seed from anything: json files, custom data, custom db, etc.  However, I can guarantee you that over 80% of the time you might need to build the data yourself. That's why if you follow the 80/20 rule, there is a higher percentage of usage, thus giving it the impetus to be a depednency.  Plus, it's a single CFC dependency, is storage an issue? Why the hesitation of a dependency?  Just go install a node library and it comes with 30!

Reply:
Seeders are different from migrations, though. I don't think we would ever need MockData in a migration, if we have the seeder functionality.  I follow what you're saying on convenience.  I just think it's "smelly", and where the mock data comes from the seeders is the responsibility of the application developer.

Reply:
I added this as a module dependency and pass in to seeds in commit 73c8bda